### PR TITLE
Query and update animals table (PR)

### PR DESCRIPTION
**Pull request by teke85**
**Learning Objectives:**

- Use database transactions.
- Modify and delete data in SQL.
- Prepare complex queries that answer analytical questions.

**Screenshots**

- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:

- [ ]  Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- [ ] Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- [ ] Commit the transaction.
- [ ] Verify that change was made and persists after commit.

Desktop screenshot

- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

Desktop screenshot (1)

- Inside a transaction:
- [ ] Delete all animals born after Jan 1st, 2022.
- [ ] Create a savepoint for the transaction.
- [ ] Update all animals' weight to be their weight multiplied by -1.
- [ ] Rollback to the savepoint
- [ ] Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction

Desktop screenshot (2)

- Write queries to answer the following questions:
- [ ] How many animals are there?
- [ ] How many animals have never tried to escape?
- [ ] What is the average weight of animals?
- [ ] Who escapes the most, neutered or not neutered animals?
- [ ] What is the minimum and maximum weight of each type of animal?
- [ ] What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Desktop screenshot (3)

Desktop screenshot (4)

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @teke85, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the readme file filled professionally. I would implore you to fill all categories of your file according to your project except the FAQ which is optional.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by teke85** [comment]
Hi @Olamarx thank you for the reminder. I have updated my readme file.

File: queries.sql [hunk 1]
```diff
@@ -1,34 +1,88 @@
...
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+/* Update all animals' weights that are negative to be their weight multiplied by -1 */
+UPDATE animals SET weight = -1 * weight WHERE weight < 0;
 
+/* Commit the transaction */
+COMMIT;
+
+/* Write queries to answer the following questions */
+
+BEGIN;
```
Comment (MaganaAsiati) [inline queries.sql]: - Kindly I suggest you remove this `BEGIN` as there is no transaction for it :+1:

File: queries.sql [hunk 2]
```diff
@@ -1,34 +1,88 @@
...
+SELECT AVG(weight_kg) FROM animals;
+
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT neutered, SUM(escape_attempts) as total_escape_attempts 
+FROM animals 
+GROUP BY neutered 
+ORDER BY total_escape_attempts DESC 
+LIMIT 1;
+
+/* What is the minimum and maximum weight of each type of animal? */
```
Comment (MaganaAsiati) [inline queries.sql]: - Please Could you ensure that your query returns the animals Who escape the most, neutered or not neutered?

    I suggest you use one of the two queries I list below :+1: 


 
```
 SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```
or
 ```
SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;

```

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 

Hi @teke85 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlonzoLeroneThisCloseGIF

## Highlights
✅ No linter error
✅ README is professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: queries.sql (hunk 1 above)
Comment (teke85) [reply queries.sql]: Hi @Olamarx I have removed the BEGIN statement

File: queries.sql (hunk 2 above)
Comment (teke85) [reply queries.sql]: Hi @Olamarx I have used one of the queries above.

**Review by prolajumokeoni — APPROVED** [review]
Hi @teke85 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - All changes requested have been implemented ✅ 
   

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**